onnxruntime/core/providers/tensorrt/onnx_ctx_model_helper.cc

```diff
@@ -272,6 +293,20 @@ Status TensorRTCacheModelHandler::GetEpContextFromGraph(const GraphViewer& graph
     std::filesystem::path ctx_model_dir(GetPathOrParentPathOfCtxModel(ep_context_model_path_));
     auto engine_cache_path = ctx_model_dir.append(cache_path);
 
+    // If it's a weight-stripped engine cache, it needs to be refitted even though the refit flag is not enabled
+    if (!weight_stripped_engine_refit_) {
+      weight_stripped_engine_refit_ = IsWeightStrippedEngineCache(engine_cache_path);
+    }
+
+    // If the serialized refitted engine is present, use it directly without refitting the engine again
+    if (weight_stripped_engine_refit_) {
+      const std::filesystem::path refitted_engine_cache_path = GetRefittedEnginePath(engine_cache_path.string());
+      if (std::filesystem::exists(refitted_engine_cache_path)) {
+        engine_cache_path = refitted_engine_cache_path.string();
```

Comment:
add verbose log message to indicate there's already a serialized refitted engine, no need to refit again.